qcengine/programs/orca.py

```diff
@@ -0,0 +1,298 @@
+
+            # Determine what SCF type (restricted vs. unrestricted)
+            hf_type = "RHF"
+            # dft_type = "RKS"
+            # if unrestricted:
+            #     hf_type = "UHF"
+            #     dft_type = "UKS"
+
+            # Write energy call
+            energy_call = []
+
+            if input_model.model.method.upper() in self._post_hf_methods:  # post SCF case
+                energy_call.append(f"{{{hf_type}}}")
+                energy_call.append("")
+                energy_call.append(f"{{{input_model.model.method}}}")
+            
+
+            elif input_model.model.method.upper() in self._dft_functionals:  # DFT case
+                input_file.append("! SP {} {}".format(input_model.model.method, input_model.model.basis))
+                input_file.append(xyz_block)
+
+            elif input_model.model.method.upper() in self._hf_methods:  # HF case
+                input_file.append("! SP {} {}".format(input_model.model.method, input_model.model.basis))
+                input_file.append(xyz_block)
+
+            else:
+                raise InputError(f"Method {input_model.model.method} not implemented for Orca.")
+
+            # Write appropriate driver call
+            if input_model.driver == "energy":
+                input_file.extend(energy_call)
+            elif input_model.driver == "gradient":
+                input_file[0] = "{} {}".format(input_file[0], "engrad")
+            else:
+                raise InputError(f"Driver {input_model.driver} not implemented for Orca.")
+
+            parallel = "%pal nproc {} end".format(config.ncores)
+            input_file.append(parallel)
+
+            input_file = "\n".join(input_file)
+        else:
+            # Some of the potential different template options
+            # (A) ordinary build_input (need to define a base template)
+            # (B) user wants to add stuff after normal template (A)
+            # (C) user knows their domain language (doesn't use any QCSchema quantities)
+
+            # # Build dictionary for substitute
+            # sub_dict = {
+            #     "method": input_model.model.method,
+            #     "basis": input_model.model.basis,
+            #     "charge": input_model.molecule.molecular_charge
+            # }
+
+            # Perform substitution to create input file
+            str_template = string.Template(template)
+            input_file = str_template.substitute()
+
+        return {
+            "commands": [which("orca"), "dispatch.mol"],
+            "infiles": {"dispatch.mol": input_file},
+            "scratch_directory": config.scratch_directory,
+            "input_result": input_model.copy(deep=True),
+        }
+
+    def parse_output(self, outfiles: Dict[str, str], input_model: "AtomicInput") -> "AtomicResult":
+
+        data = cclib.io.ccread(io.StringIO(outfiles["stdout"]))
+
+        properties = {}
+        extras = {}
+        extras["ev_to_hartrees"] = 0.0367493
```

Comment:
Is this from Orca itself? Otherwise best to use `qcelemental.constants.conversion_factor("ev", "hartree")`